### PR TITLE
Improves external IP idempotency

### PR DESCRIPTION
**Pull request by bnaecker**
- Fixup in the external IP actions in the instance creation saga. We
  added nodes to create UUIDs for both external IPs and NICs, but only
  used them in the latter. This uses them in the external IP case too.
- Make the external IP allocation query idempotent and add test
  verifying that.

### CODE REVIEW

**Comment by bnaecker** [comment]
Resolves https://github.com/oxidecomputer/omicron/issues/1316

File: nexus/src/db/queries/external_ip.rs [hunk 1]
```diff
@@ -887,4 +933,70 @@ mod tests {
...
+            )
+            .await
+            .expect("Failed to allocate instance SNAT IP address");
+
+        // Check identity, not equality. The timestamps will be updated.
+        assert_eq!(ip.id, new_ip.id);
+        assert_eq!(ip.name, new_ip.name);
+        assert_eq!(ip.description, new_ip.description);
+        assert!(ip.time_created < new_ip.time_created);
+        assert!(ip.time_modified < new_ip.time_modified);
```
Comment (smklein) [inline nexus/src/db/queries/external_ip.rs:991]: <= ?

**Review by smklein — APPROVED** [review]
LGTM; thanks for doing the extra work to support idempotency!

File: nexus/src/db/queries/external_ip.rs (hunk 1 above)
Comment (bnaecker) [reply nexus/src/db/queries/external_ip.rs:991]: Yeah, I was on the fence about this. Strict inequality is fine for test though (1) it's not really guaranteed and (2) we don't care so much in real use cases. I'll update.

Comment (bnaecker) [reply nexus/src/db/queries/external_ip.rs]: Done in 5292bdab

**Pull request merged**